skimage/color/colorconv.py

```diff
@@ -801,12 +801,8 @@ def rgb2gray(rgb):
         return np.ascontiguousarray(rgb)
 
     rgb = _prepare_colorarray(rgb[..., :3])
-
-    gray = 0.2125 * rgb[..., 0]
-    gray[:] += 0.7154 * rgb[..., 1]
-    gray[:] += 0.0721 * rgb[..., 2]
-
-    return gray
+    coeffs = np.array([0.2125, 0.7154, 0.0721], dtype=rgb.dtype)
```

Comment:
Why `rgb.dtype`? Is `rgb` guaranteed to be a float type?

Reply:
yeah, the prepare function makes it a float of the correct range.

I feel like removing that would be considered "premature optimization"